python/lsst/daf/butler/tests/testFormatters.py

```diff
@@ -62,3 +67,13 @@ def validateWriteRecipes(recipes: Optional[Mapping[str, Any]]) -> Optional[Mappi
             if "mode" not in recipes[recipeName]:
                 raise RuntimeError("'mode' is a required write recipe parameter")
         return recipes
+
+
+class LenientYamlFormatter(YamlFormatter):
+    """A test formatter that allows any file extension but always reads and
+    writes YAML."""
+    extension = ".yaml"
+
+    @classmethod
+    def validateExtension(cls, location: Location) -> None:
+        return
```

Comment:
I spent an hour on it and my main problem was trying to work out whether I deliberately used a mix of YamlFormatter and JsonFormatter or if they just happened to be the remnants from one of the other tests and we never noticed before because the format was irrelevant. This triggers solely in the chained datastore testing of constraints on ingest where there is a specialist datastore that overrides a formatter based on the dimensions rather than storage class.  I will take another look at it but I don't think I consider this to be technical debt as such -- it's a side effect of the complex testing options set up for one datastore. If I've dealt with everything else and this is my only remaining issue and I'm able to get it in the weekly today I'm inclined to merge it.

Reply:
Having already spent an hour looking for an easier solution is definitely enough due diligence for me.